### PR TITLE
Use 'es6-promise' for all Promises (closes #71)

### LINKED ISSUE #71: Use `es6-promise` module for the promises
It's necessary to use the same module with the `Promise` class on the client and server sides. The `promise` module have no implementation for the client-side

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -40,6 +40,29 @@ var UI_STYLES_FILE_PATH = '/client/ui';
 var UI_STYLES_TEMP_DIR  = path.join('lib', UI_STYLES_FILE_PATH);
 
 
+function promisify (fn) {
```
Comment (inikulin) [inline Gulpfile.js]: There is no need for this code duplication. Just use Promisify from utlis here

**Comment by inikulin** [comment]
\r-

File: Gulpfile.js (hunk 1 above)
Comment (AndreyBelym) [inline Gulpfile.js]: `promisify` from Utils cannot be used in `Gulpfile.js`, because it must be transpiled from ES6 to be used from ES5 code.

Comment (inikulin) [inline Gulpfile.js]: Let's use https://www.npmjs.com/package/es6-promisify then. This covers both cases discussed here. We need to replace it in hammerhead/browser-natives as well for consistency, please fill the issue where.

**Comment by AndreyBelym** [comment]
FPR

File: Gulpfile.js [hunk 2]
```diff
@@ -171,7 +172,7 @@ gulp.task('build-styles', ['styles-temp-copy'], function () {
 
 
 //Build
-gulp.task('build', ['lint', 'wrap-scripts-with-templates', 'build-styles'], function () {
+gulp.task('build', ['wrap-scripts-with-templates', 'build-styles'], function () {
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: why the `lint` task is removed?

Comment (AndreyBelym) [inline Gulpfile.js]: oops, removed to use console.log for debugging purposes, forgot to get it back

**Comment by AlexanderMoskovkin** [comment]
\r-

**Comment by AndreyBelym** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm @AndreyBelym Please, fill the issue to other modules to switch to es6-promisify

**Comment by AndreyBelym** [comment]
Note on `es6-promisify`: we've discovered a difference between it and `Promise.denodeify` (and our own `promisify` implementation from hammerhead).

If node-style function has callback with conventional  two arguments `(error, data)`, everything is identical: Promises returned from both `es6-promisify` and `Promise.denodeify` resolve `data`, and reject `error`. But if callback is called with more arguments, like [request](https://www.npmjs.com/package/request) - `(error, response, body)`, then function promisified with `es6-promisify` will resolve array `[response, body]`, unlike `Promise.denodeify` and our own `promisify`, that resolves just `response`.

**Comment by AlexanderMoskovkin** [comment]
@inikulin Is there a reason to do it in the `testcafe-browser-natives` repo? We have no any client code there

**Comment by inikulin** [comment]
@AlexanderMoskovkin promisify is used only on **server** side.

**Comment by inikulin** [comment]
@AlexanderMoskovkin Oh, I see now. We still using promise lib there. Overall, it would be nice to switch to es6-promise there for consistency.

**Comment by AlexanderMoskovkin** [comment]
ok

**Pull request merged**